tests/unit/style/rule_tree/bench.rs

```diff
@@ -73,13 +73,29 @@ fn bench_insertion_basic(b: &mut Bencher) {
              .bar { height: 500px; } \
              .baz { display: block; }");
 
-        for _ in 0..1000 {
+        for _ in 0..(4000 + 400) {
             test::black_box(test_insertion(&r, rules_matched.clone()));
         }
     })
 }
 
 #[bench]
+fn bench_insertion_basic_per_element(b: &mut Bencher) {
+    let r = RuleTree::new();
+
+    let rules_matched = parse_rules(
+        ".foo { width: 200px; } \
+         .bar { height: 500px; } \
+         .baz { display: block; }");
+
+    b.iter(|| {
+        test::black_box(test_insertion(&r, rules_matched.clone()));
+
+        unsafe { r.gc() };
```

Comment:
Might be worth wrapping the RuleTree object into a struct so that you can call its `gc()` in a `Drop` impl, to save repeating this in each test.